HPCC-17044 Update DZ configuration structure

rpastrana opened the pull request:
- Removes computer attribute
- Renames WhiteList to ServerList
- Hides name entry

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)
-[x] This change forces configuration changes

## Checklist:
- [x ] My code follows the code style of this project.
  - [x ] My code does not create any new warnings from compiler, build system, or lint.
- [x ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [x ] Scalability
  - [ x] Performance
  - [x ] Security
  - [ x] Thread-safety
  - [ ] Premature optimization
  - [ x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@xwang2713 please review. Please take a deep look at the changes to  initfiles/etc/DIR_NAME/environment.xml.in

HPCCSmoketest commented:
Automated Smoketest: :x:
Sha: b5b73b29508ad78df5d610316e47c9b09f48feef
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 90.52 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: Fail
HPCC start:
Exception: 2:
Error - syntax error Mismatched opening and closing tags [line 26, file offset 804]
=ServerList server=localhost
  </D*ERROR*ropZone>
  <RoxieCluster allFilesDynamic
There are no components configured to run on this node ...
HPCC status:
Exception: 2:
Error - syntax error Mismatched opening and closing tags [line 26, file offset 804]
=ServerList server=localhost
  </D*ERROR*ropZone>
  <RoxieCluster allFilesDynamic
There are no components configured to run on this node ...

HPCC Stop: OK
HPCC Uninstall: OK

AttilaVamos commented on initfiles/etc/DIR_NAME/environment.xml.in:
```diff
@@ -17,12 +17,13 @@
             directory="${RUNTIME_PATH}/mydropzone"
             name="mydropzone"
             ECLWatchVisible="true"
             umask="022"/>
```
I think this closing tag is wrong, because it closes the <DropZone.

rpastrana replied:
Good catch.

jakesmith commented:
@rpastrana - are you still aiming this for 6.4 ? 

From what you were saying yesterday, the UI changes would take too long, so would have to pushed back ?

rpastrana commented:
@jakesmith this is the pull request we talked about for a long time with an ultimate decision to go forward because you felt the eclwatch/esp changes would be in place for 6.4. Are you asking to back out this change now?

rpastrana committed:
HPCC-17044 Update DZ configuration structure

- Removes computer attribute
- Renames WhiteList to ServerList
- Hides name entry

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 0ef30ddc086904a9109d0f4fac8aaca636505701
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.162 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
@rpastrana - change looks good.

But we can't take this change without dependent Esp UI/service changes (I couldn't see an issue opened, so just opened one: HPCC-17084)
If that can't make it into 6.4, then this PR can't either.

AttilaVamos commented on initfiles/etc/DIR_NAME/environment.xml.in:
```diff
@@ -17,12 +17,13 @@
             ECLWatchVisible="true"
-            umask="022"/>
+            umask="022">
+            <ServerList name="ServerList" server="localhost"/>
```
Is "localhost" really can be used as a server address? How this can be resolve in multinode system?

jakesmith replied:
It wouldn't make a lot of sense in multinode systems, but it does on 1 node system.
However, we tend to use ".", meaning 1st preferred NIC ip, which would avoid this issue.

rpastrana replied:
This is the default config for one-node cluster.
@jakesmith good point.

rpastrana committed:
HPCC-17044 Code review change

- Changes default env.xml server value for DZ from localhost to .

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@jakesmith please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 978dc8c188e453eb85bd972e373001591b3cb3d5
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.555 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
@rpastrana - looks good.

@richardkchapman - ready to merge.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 978dc8c188e453eb85bd972e373001591b3cb3d5
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.931 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
Waiting to see if HPCC-17084 ends up mergable...

rpastrana commented:
@richardkchapman this one can be merged now. As I understand, Attila's recent PR relies on this change.
